src/backend/distributed/commands/statistics.c

```diff
@@ -359,9 +359,19 @@ AlterStatisticsSchemaStmtObjectAddress(Node *node, bool missingOk)
 	AlterObjectSchemaStmt *stmt = castNode(AlterObjectSchemaStmt, node);
 
 	ObjectAddress *address = palloc0(sizeof(ObjectAddress));
-	String *statName = llast((List *) stmt->object);
-	Oid statsOid = get_statistics_object_oid(list_make2(makeString(stmt->newschema),
-														statName), missingOk);
+	List *statName = (List *) stmt->object;
+	Oid statsOid = get_statistics_object_oid(statName, true);
+
+	if (statsOid == InvalidOid)
+	{
+		/*
+		 * couldn't find the stat, might have already been moved to the new schema, we
+		 * construct a new stat name that uses the new schema to search in.
```

Comment:
This logic seems slightly dodgy to me. If the object exists in the new schema, prior to ProcessUtility, it would return an address?

Reply:
It sounds right but wouldn't standard_utility fail the transaction in such a situation? 
By the way, we have that logic a couple of more places in our code.

Reply:
I also agree that this is not the best way to approach this problem. One alternative we discussed with @aykut-bozkurt in general is that: Should we split `address` function into two functions: `PreprocessAdress` and `PostProcessAddress`.

With that, we can also ensure that, the generic `address` function shouldn't be called during postprocess of `DROP` because that object does not exists anymore. We currently implicitly never call `address` on `DROP` post-porocesses, but could easily be overlooked.

However, that looked like a pretty large refactor, and we didn't want to get into such a refactor.  Do you suggest going into that direction? Or, any easier approach can you think of @marcocitus?

> It sounds right but wouldn't standard_utility fail the transaction in such a situation?

Can you please try that? I think adding a regression test sounds very useful to see we are NOT breaking anything here.

It'd be very useful if you can consider adding tests where we have this pattern in the code.

Reply:
I think it would be nicer to just have an additional argument (e.g. `bool isPostProcess`)

Reply:
and, @aykut-bozkurt will try to get rid of the hacks in the code that is similar to what we do here in a separate commit.

Reply:
I introduced `bool isPostProcess` as address function parameter and used it to resolve the problem. It will also help a couple of more such cases. It can be also useful for future things as well.

I closed that PR, and moved related changes to PR #6088.